workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/Footer.tsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+
+import infoCircleIcon from '../../../assets/icons/info_circle_icon.svg';
+import {getSiteURL} from '../../../components/InviteMemberModal/services';
+import {Liferay} from '../../../liferay/liferay';
+import {paymentMethod} from '../enums/paymentMethod';
+import {StepType} from '../enums/stepType';
+
+interface ProductFooterProps {
+	addresses: BillingAddress[];
+	cartId?: number;
+	enablePurchaseButton: boolean;
+	handleGetApp: () => void;
+	isFreeApp: boolean;
+	sectionProperties: SectionPropertiesType;
+	selectedAccount?: Account;
+	selectedPaymentMethod: PaymentMethodSelector;
+	setStep: (nextStep: StepType) => void;
+	sku: SKU;
+	step: StepType;
+}
+
+type SectionPropertiesType = {
+	[key in StepType]: {
+		backStep: StepType;
+		nextStep: StepType;
+	};
+};
+
+const ProductFooter = ({
+	addresses,
+	enablePurchaseButton,
+	handleGetApp,
+	isFreeApp,
+	sectionProperties,
+	selectedAccount,
+	selectedPaymentMethod,
+	setStep,
+	sku,
+	step,
+}: ProductFooterProps) => {
+	const getButtonText = () => {
+		const isAccountOrLicenseStep =
+			step === 'account' || step === 'licenses';
+		const isPayMethodSelected = selectedPaymentMethod === paymentMethod.PAY;
+		const isTrialMethodSelected =
+			selectedPaymentMethod === paymentMethod.TRIAL;
+		const isOrderMethodSelected =
+			selectedPaymentMethod === paymentMethod.ORDER;
+
+		if (isFreeApp) {
+			return 'Get This App';
+		}
+		if (isAccountOrLicenseStep) {
+			return 'Continue';
+		}
+		if (isPayMethodSelected) {
+			return `Pay $${sku?.price} Now`;
+		}
+		if (isTrialMethodSelected) {
+			return 'Start Free Trial';
+		}
+		if (isOrderMethodSelected) {
+			return `Create PO for $${sku.price}`;
+		}
+	};
+
+	const onCancel = () => {
+		Liferay.Util.navigate(getSiteURL());
+	};
+
+	const onPrevious = async (previousStep: StepType) => {
+		setStep(previousStep);
+
+		return;
+	};
+
+	const onContinue = async (nextStep: StepType) => {
+		const isAccountStep = step === 'account';
+		const isPaymentStep = step === 'payment';
+		const isLicenseStep = step === 'licenses';
```

Comment:
We have these values being repeated a few times in the code. I think it will be better to create an enum for this.

Reply:
Well noted! Thank you!